src/Approximations/overapproximate.jl

```diff
@@ -1636,6 +1647,31 @@ function overapproximate(X::LazySet{N}, ::Type{<:Zonotope},
     return Zonotope(c, G)
 end
 
+function _overapproximate_zonotope_cpa(X::AbstractPolytope, ::Type{<:Zonotope},
+                                       dirs=OctDirections)
+    n = dim(X)
+    p = Vector(undef, Int(floor(n/2)))
```

Comment:
can we concretely type `p`?

Reply:
.. if `project` uses dynamic dispatch (VPolytpe/VPolygon/Interval) then it gets more complicated. that's the reason that i find it useful to have type-stable internal functions as suggested [here](https://github.com/JuliaReach/LazySets.jl/pull/2287#issuecomment-665160969).

Reply:
Yes, that is the problem, when you project a `VPolytope`, it gives you a `VPolygon`

Reply:
hmm ok but the blocks are all of size 2 here. then we can use `p` of type VPolygon if the system dim is even and a union of VPolygon and interval if it's odd.

Reply:
That's a good idea! I will do that.